packages/nextjs/src/utils/instrumentServer.ts

```diff
@@ -346,12 +350,27 @@ function makeWrappedMethodForGettingParameterizedPath(
 /**
  * Determine if the request should be traced, by filtering out requests for internal next files and static resources.
  *
- * @param url The URL of the request
+ * @param path The path of the request
  * @param publicDirFiles A set containing relative paths to all available static resources (note that this does not
  * include static *pages*, but rather images and the like)
  * @returns false if the URL is for an internal or static resource
  */
-function shouldTraceRequest(url: string, publicDirFiles: Set<string>): boolean {
+function shouldTraceRequest(path: string, publicDirFiles: Set<string>): boolean {
+  // Don't trace tunneled sentry events
+  const tunnelPath = globalWithInjectedValues.__sentryRewritesTunnelPath__;
+  if (tunnelPath) {
+    const startsWithTunnelPath = path.startsWith(tunnelPath);
+    const transactionNameRest = path.substring(tunnelPath.length);
+
+    // Check if the requested path matches the path we tunnel requests to
+    const restMatchesTunnelStructure = transactionNameRest.match(/^\/o\d+\/\d+\/?$/);
```

Comment:
M: Is there a reason not to just compute the actual value and match against that? 

(Semi-answering my own question: I suppose people could be using different DSNs for the front and back ends, and the backend wouldn't know the front end's DSN... If that's the concern, can we note that in a comment?)

Reply:
The only thing we really care about here is the pathname of the incoming requests since users define the pathname to route to via `tunnelRoute`. I refactored this PR to use query strings instead to be a little bit more predictable in where proxy requests will go, just from looking at the `tunnelRoute` option so this logic right here became a lot more simple anyways.